csrc/kernel_cache.cpp

```diff
@@ -770,17 +840,38 @@ std::unordered_map<Val*, const ArgAbstract*> FusionKernelRuntime::
     // the original tensor input. See note [Trivial Forwarding]
     for (const size_t group_out_i : c10::irange(group_outputs.size())) {
       if (!group_outputs[group_out_i]->isFusionInput()) {
-        args.push(group_runtime_outputs[group_out_i]);
-        tensor_map.emplace(group_outputs[group_out_i], args.back());
+        fusion_args_.push(group_runtime_outputs[group_out_i]);
+        tensor_map_.emplace(group_outputs[group_out_i], fusion_args_.back());
       }
     }
-  };
+  }
+};
+
+} // namespace
+
+std::unordered_map<Val*, const ArgAbstract*> FusionKernelRuntime::
+    runSegmentsWithInputs(KernelArgumentHolder& args, bool is_dry_run) {
+  TORCH_INTERNAL_ASSERT(
+      args.size() == segmented_fusion_->inputs().size(),
+      "Inputs were not set up correctly, received ",
+      args.size(),
+      " inputs but expected ",
+      segmented_fusion_->inputs().size());
+
+  // map from val to args
+  std::unordered_map<Val*, const ArgAbstract*> tensor_map =
+      mapFusionInputsToArgs(args);
```

Comment:
Perhaps this should be also moved into `ArgumentManager`? Then we don't need to keep `tensor_map` here, but it can be just a member of the class, right?

Reply:
moved.